pkg/services/store/object/object.proto

```diff
@@ -5,41 +5,45 @@ option go_package = "./;object";
 
 // The canonical object/document data -- this represents the raw bytes and storage level metadata
 message RawObject {
+  // Grafana resource identifier (includes kind+uid)
+  string GRN = 1;
+
   // Unique ID
-  string UID = 1;
+  string UID = 2;
```

Comment:
I am torn if we want to duplicate the ID+kind as explicit fields.

Reply:
Wdyt about these two alternatives

1. have just a _GRN-as-a-struct_ field (ie. https://github.com/grafana/grafana/blob/49214a6f066341174c964183d39b9e2657ffc876/pkg/infra/grn/grn.go#L9-L22)
   - the field could be just named `grn`
   - this has to come with an SDK method that takes the _GRN-as-a-struct_ and serializes it
2. have both _GRN-as-a-struct_, and _GRN-as-a-string_
   - the fields could be named `grn` for the string and `grnInfo`/`grnDetails`/`...` for the struct
   - easy access to both the string version and the info, makes it super easy to work with the API for non-golang clients without access to the SDK

Reply:
I think it is important to group the `uid`/`tenantid`/`resourceKind` under a single `grnInfo`/... property to make it super obvious that these fields offer no extra information, its just the parsed version of `grn`

Reply:
ya -- sleeping on it, I think you are right.

I avoided this originally since the string flavor seemd like a simple/independent way to address items, and I am not wild that the proto based GRN is always passed as a pointer.

That said the values split out is really what is needed and more clear so.

I'll take a stab at having a GRN struct in protobuf.